### PR TITLE
MF-1055 - rollback/release transaction on error

### LINKED ISSUE #1055: Postgres writer problem with not cleaning up connection
when posting  data into postgres which is byte array 

```
mosquitto_pub -u $THING -P $KEY -t channels/$CH/messages -h localhost -p 18831  -m  '[{"bn":"1:", "n":"exec", "vd":"�����Y�q�PA\u0017B�\u000bQtU���hM/I/16�I[5��6IӦ�\n\u0005)�P\\�XE�\u0011Jq���JqgAAT\u0014*�\u0010�TDŅ�F\u0010� ��2�y��7�|v������3�m��\u003c.�Fa��[�2wR�Qy4�s�ie�{2�]�p϶\u001b+(\r��\u0014��i~\u0015��=Y\u0019f-D���S\u003e@�7���[\u000bV��^��{敏����\u0004k�ۿ⏫��\u003ep�,�ډ\u001f����z1�O�C�a*��_w|m��O۟�)ۀ�H��{��ғ�{ϔ\u003e��%�Ꞣ�R��?��{�S�}N\u0019-��Ӕiړ���\u003cq�\u001b��WO\u0011gƊ)��\u001b�3c����k΂0��c{��\u0019�$έ����ǈ?c\u0001mW\u000fp|k�+3ԃůw{���4yB���!\u003e��O��q=�O{��3�A��\u003cUۭ����S�_������o��˨\u000f����O�E��z#�Uȿ\u0015�e=\u001f���4��^�Uۭm���j�����k*\u001f������D�����\u000c�0iX=R�e\\��7\u0016��Ҍ�����蔧����z�z��$������\\ǗG]�z���\u0013~�\u0004���\u003c�����9��V�J�\u0015�f����3�w߅O��_��U�Q|I\u001f�6�\u000f�\"�\u003e�/�_�U�����G�w|�Q�ø�������{��}�Y�\u0011׬m�9�?�9Vwk�lq�9X�B��z\u0017�/OI�ꩥǔ��v}���\u001a;�߸\u000e?�C\u0014\u003e�C�\t�Q��c��������å_e����V�h��2�3���̑\u003eݿq\u000f¯r���K��U�Ŋ�W\u003eخ�������\u001e��5Ȕ/��\u0008�ʣT����\u001f!�\u003ez\u001e\u0017ۃ0�|{m���s��u\u000e�_Vۃ�ڛ�G#�ȃ���-O;�2O{�4�mC�{���+��yc�ďy^�������l��I_�o����\u0007f\\��7�O�kn��S�t}����ڎ�\u001f�\n\\��tί=�ʤ\u0018��=��\u0014g���6ck��_���mw�?�O\u001aTF���ES�ך����\u0000u?§jpi۲�3]������Oy�v�Qҿ2U��n��3VU�,��i����wXėy�g�1���i�\u0011���:���=�v����\r~�W{����J\u001f�\u0014�G{�g^��\u001f�h�\u0006��9$Ϲ\u0006eR�q��񫾪oL�3��Ҵw'q�\r�sT\u001f�0��m�]���S�t�Ԓi�_|�3VJ�Tݰ���\u003c�;��~���O{���\u0007�̩�e��\u0008?e��3OS������\u001eZ��ʽ�7\u0005��ҟ���#c�\u003cS�m�??��\u001e.�����x\u000f\u0017���ʟ���\u001f;���wm�}\u0014������Y�dn�����'���\\�qe�T�w|�\u0014iJ�)M�������҄�X����f�ߚ�%3���|��=��I��ϸ\u000e篚��\u0003|o�?e��j����'�k���\u0010\u003e�L�c|��Ӈ���~�\u000f��\u003e��9�u\u001b]��\u0013�S�R�~ڮğ�\u001c��:���%s{�էL){�6zyλ���2�����\u0007��}/P�\u0019w!̥\tq��Mg�\u000cs:��gc������t�VKs�׻���Z��z�\u003eﾮO{���n.O�=Ϻ����aQf+��\rٿj�T�K�W\u0026�\u000f�\u0013��\u0007eJ���\u0017%3�_�j:��s?��י��+g�{��=\u0013��)�]�w/iV{�\u003c�_S�W\r��C2�A��\u001e,L̝�f�\u0007P�b�f�W{��+�Qտ����I�Sq'�]$O)3܏�J�_yn\u0015j�\\��{�\u003c�=Y�w�ҿ���:���wX�\u000f۵\u001d�s�mW�����7as��/��_}��G��z\u001a��b)�'8~��RoJ��)��\u0003�\u0019�\u0001a�=T��;)�\u000eį�����7\ns�\u0001��o�8�� ��k�ĩy���7\u003cݟu�\u001c��f�����=N��v��w|e����_z��\u003ee��'����\u0019\u000b�\u000c�\u001e���ٮ_=L��\u001e'}VΊ:��k\u001e��+=�X�2F���\u003e~�\u003c�c{ťX���\u001b�䙠�������?��\u0019�(�?~�.�4!L��Y�\u000bũ㫧�#��ܳ\u003e\u001e�ln�\u003cU�b�?�\u003e뺹?e�8�7\\�Qi�����I��LV�h�\u0011\u003e�\u0013��p��[W��y���ޥ.\u001f��2�Y���_�����j�c�g�\t_��ҏ���_�=��P\u003e��3�LS�\u0008\u000f��\u001e���ǻ��k�6k�!SFW�\u0016� ���2�R��=��ii޻�01�Z�����k�w�ƅH��\u0008��Ҽ�{�_��ܿ�$į��~e��\u0007��ܟ4o�\u0010�\u001b�#}�{u�hܟ��ſg�2Q����M\tO}��W�_�G����q1��jx\u00197���W�z�?�y��H\u001f�\u0019+��a������J��\u001d_۱�\u0017���\u001dأ�3�Y�\u0013=G�\t�\u001d#�ZKP����\u003e\u0006�\\yp��*Δ�~����z^�\u0008��\u003e\u0018��~�Y��³bݬ'\"��F�y����Twou\u0002�������S�9yLx��-=[�k�\u0013��\u000el7����7�?��v�+i\u001bp���f�M��=�3Sۛ��SI_�.z�\u000f�^O�K�o-�)�͝\u0014��Z9��lҰz�0+\u001f�4+Oj;�f�#s|���lٳ��R��~�G\u001ep=�������I�d�W룕猅������(RO\u0016\u001e�y��3Y������\u003cS���\u0015�#Ŀ������\u001b����es����g�d�O�I���w���(��GI_%ӭ�b{�o+�ğos�c�\u001c;x�E���o��\u0013�����2�7\u0016���'!Mx��s��+\u0013̹\u0010g�\u0001�W̻2K�s=�ϻ�2Q�\"|�n��'O�\u003ee�5��_{���#��oBH��LyV{����#��ㅇok9��\"�̳���V\\ݽ�[\u0011�ҧ�{\u000f�L�\u001e�̪w��\u0015\u0015\u003e�#�~�/~ʇk����������_z�~���\u001d�\u000b.̬\u0005��4����t�_�U\u0026X\u0017L�k�\u0014f�g;�\u003e2ף���\u000cק�@�\u0018�Rz���\u001aP�#�_�+㗕\u0019���)s煏�\u0010���X�\u0019_�L�^������Qf�\u003e����w���g��Ӈ-|��W\u0026Z?����X�7��[s_��\u001e���_�\u0019+�=��S��zO�{���̳�z����u���.�9������0�g�U�L�7h�;�}�vy���{�S��Ҕ\u003e\u0006iƸ\u000fq������Wہ��C=���e�ӟQ�\u000b�+�\u001c߸\u0011����L3���ګ�\u0007������K��cհҟc�g��g\\��\u0017?҇oNخ=_�;��/}�?���G��\u003eu\u001e�\u0014k@)\u001f�ْ\u003e�ې��^���)�}�C�Rf�nq.\rJ3�C��\u0012�����ނwCqj�����O�*������_qn���{N;�9\u0011��{�4i|��ч����I_ڞ���_V��\u0013��=ôgK_֫��g��\u003ef���c��\u000fM�냷�Z�~�_K�Ɏo�u�K/~U^۞��\u001e%��!�\u001e�\u000c���.���ؿl�SFۧ�\u000f{���e����T�S�]����!$L�������9oN�\u003c�\u000f����o\u001e���6 �T�\u0015�\u0010�7\u0016�3��ҲG���\u001f�V�����_���މ\u003e$���N��=�\u003c�\u001d_x�~�D��OQ�Yײ����u\u0005]��a:�oR8ŊZ+X�k�w?\u0015��mR�j�q}��Զ��\u0007�I�\u0019����,�Q��{-��C�Y�K���s�^�yDί\u001ef�wg�\u0017\u003e�O{�4U��\u001b\u001f��i��2U\u001f�_�\u001ch�p|iV��������\u0013Tw{�o~�g�9=�7�����h�_=�\u003ca���_�Ǹ\u000f��\u0019�y�;�\u003eT�k[s��]��v���-}H�/���;��\u0008�+/�ߔѵf�p\u000fu�wM�\t*��St�zcV��z�ia�=W�#��)��W�Dm/�_~{�\u001e���\u0003�\u0017�\u0019�'OYk��{w�~��\u001c�r�}���W]\u003c�\\�g�3�9��b�\u001c_�5��3��u��1��j/xϴ��r\\���m7�L�X�Kx\u0019\u000b(}�\u000f\u0010�U�������۞Y�Xm�v9et�\u0004���\u003c��A�\u0019�Vz���\u001b���mW\u000f�LО.̪�r�=�I����4o�R���\u0019�O���s�ʭ���+��\u003cԺ\u001b*�J\u000f3�����l�)(S�{���\u000c�1�O�t�ړ���왦�T��mI�X\u0013��)\u0013]Ո2VR��� }{��=�v߹�?�g�_�(��\u0011��)��\u0019yZ�Q��\u0026�]��_\u0019�O�[����?��ٲ��G\u0014N���\u000b���9���Ҵ�*q��E��=�z\u0011��z\u0005��J��\\ק�K������s��������wd�?�q�Oۊ�����c\u0015\u003e�k\u001bp}�6��kP\u0026zOv�\u0014��3�?�����,��=���\u000f\u0011���{�iOW�����������:��L��5Q�\u0015ϋ\u0013i�sȜ\u0017y^�s|�����)�|#����ʬ�g8�힩Ҭ�Ǟ9�������LP�V\u001e��nyF���3����et��\u0007+3���ô\u001e��+\u0013ů�n����\u001b��C����ҷ:���=V���s�\u0014�;����ď9q�w�x��p|��V\u0026�ӈ�'\u000e��2\u000fP��ñ���q\r���SiF\u001f�{R�+3��\u0014���;~ك�s:k�v~�D�4qn����Ɣ8w|�/~�_�\u001f�m���e��G\u0026�+��?�(��z ҷ�\u0017�C{����������ܟ97�W-L�P��Z��k��m(k\u000c�\u001f�å���ԟR�\u0010����~mS�O{��kΘ��.����ޙz�0���0+��9#εGKs�\u000b�\u0003�m\u0026s;�\u0019�\u0008痦�i��*�B�\u001b���=��\tc��\t�.ܿ���_ۈ�מ.~�����z�Y��\u0018?�k\u001b��W\u003e�z��X3�Wo$����,~�\u0017��g��Y�V�\u0015¯j������\u0015\u0015\u003e�ܪס���{�r~�\u0017?�$��3��O\u0019-̪\u0026}�\u000b�E�\u0026���.N�\u0019֨�\u0026]�{�\u001d\u0004iR{��\u001b+��#�y�ͥ�E�\u001a��~��U\u000f0~���gٿb\u0001�\u001b\u0010~�gk�+��7;~�)��?�\\�k]5y�\u003c5��=\\������%���\\u\"e�\u003e\u003e�S���y\u000f���f��=A�W�F�O/~�n�?�G���M�������\u001f̙�2Z{�4cΊsx��3�L�QO\u0012����/��s�n\u001b\u000b$�\\�4m=\u0012ׯ=X��L+=D�*���9#�Ǹ\u0003iR�q�U8���җ5d�)�Z�\u001f� yB=������sҿ~y�_�a�Ó\u003e�yҿ�\u0004��O����k�*3�/}+?\\�\u003eĲW�_mO�k��\u003e�_��{���N)~��\n���\u003e�]vr�\u001a1����\\S��瞄��\\eϵ]=D�\n\u0013a�~��\u001d�S҄\u003c�\u001e(���W�v��]����~����~�+��U��z0ҟ�fe��'�*\u003e^\u001f���\u001e%��%s����\u003eb��{{~�G�\u000fT��L��\u0018+\u003e�d��\u001fej�\u0011��)���\t\u001f�\u0014q����_'���y���X\u0015s���6v��Q�9�̨X\u003c�Yi���Ҭ������\u0013+�L�X\u000bQ��n*~�\u0005s~�ݟ�V��L\u0012?���������u\u003c}��_Պ5�B�1g��y\u0026*s�=\u003c�'�WNR��6\u0016��\u001b�+\u001ak����J_�?�]��㗿���2�vu��\u0007~�\u001d����`=\u000e�/~v:Ӂ\u001emN�{PFI3�\u001f\u000b�k��/Ϻf�A\u00083}(�Lۊ�7.������������Ұ\u003eXe��*yZ=�kۨ{�z��n��=M�V}a����zOѻ���w���4�Ϝ\u001a�Wo(��Y�vK��\u001d\u0017�7O��[wO�i{���\u001b�������Ý_�\u0011�ֈ�?ʟSy��0H?���\u003c�{�W��ho���0�w���S����zC՜\u001caZ5�}k���\tiλ����\u000b�z�H{|�T\u001d��\u0011��v|c�=�+֯ꗻ~����.�\u003e��#�\u001e���i���$�^/��)��k\u0012��3�\u000f�H\u001d߻��+\u001f���\u0018\u0001���\u0026}x�p~����s=�ŸuNǟ2Z����g��sϞ��L=C�W�-홮�3A�Y\u0003�Ϛ��o�\u0007�o\\��3�M��C�_5�*Ϭ�S=�l7O��r���!���U�y�����\u0011�3���L����]�g��n��ov�DI/���(�\u001c�X�¯�aΈ�m���1�H��m;�K��]{�:Yٳ��v}�m�\u001fs�\\�����K���G(ε';�q\u0007Ҭw{?}4��wci�| �Xl�o���X,q�U~��{��k�t~�\u0014�W=��W���3S{��3OU�+��i���_o�UN�2I���\n?�Y�=���s�~|{J�T�$u*۫v��R��L\u0013�\u001f���L�����G����'���e�Q�mL�V��?kL�.�I_O\u001d�\u000b^5���Ksګ�I�B�=P�4\u003e���\u001eܟ2Q��\u0013�o�x�t۽���_\u001f��viT��?c�9�5��lc\u0001�_�d�����(�[�\u0003���?�0��\u0013t|e��+{��v��\r~�o\\�z��\u0012�W��]\u001f�����O۝�����91�׺��dc\u0008�mW��G왦|�\u001f��\u0017?�c�����������s��7��5�gJ3�=��[��\u0014i�s�򰼛�9[w?ׯ�B�0'B�R����Xzj�h��j������q���Io�W{���\u0003���ky�3��`�I�\u0017\u0018����M8�:����»�1����\u0003�o܆��mΜ:�Wy�ޣ���\u0001\u001c_ۓ�6���M\u0019-N˞%��\u0013��b��L��z\u000cҬq\u001f\u0015�k{��4.C�x\u0026H\u0013憺���\tߪ����)���Ds\u001e���\u0008��z��3��}�Y��;\u001d�Ņ�\u003eby�{@�\u0006�\u000f�g�o���v(|�E��j���*\u003e�O}`ί�K���\u001fׯ��M\u0005�W�\u0017\u0015W�?Fx�S�}��/\u001f����|�h�h�����uxwt��ДI�����8�n%Mx�u}��]��La�\u000fH�{7���6 ��l7�U��\u000cq�U\u000fB=G�k�v\u003e�Lۭg�����߻���\u0007.���¯���\u000ep}�����\u0004�\u001b�%��OP\u0026�\u0007j�z�E\u001f���\u001e.�\u001a\u0017\u0026}�\u001b���\u001d_�ڿ�\u001az����1�����ɱ�u%���A�;Ǵ�\u003c8e��F�̸\niB����:I�U������9���|�,ۭ����G�Qu��\u0005+�\u0016ͫ�J3�Q�_�\u0016\u0015_]q\u000b�:�.�9��R��2��U\u001e��*�i�d�#O��]�K��O=��i���ʶd�ke�z�2���T,���\u000b?�#��M�\u001fk#�\u001f�\u003e\u001c�3��k{��v�)��C���{�\u0013eB�N\u001c�X\u001bqj�i���9�\u0017f�}�i�TiB=Z��\u001e��ў-|�n'O�~���\u003c}\u0018u7�����_\u001f����*}�� ��)\u001c_\u001ep����o�\u0010�X:���mGx;�g��Q�j�pǷF��W���\u0013\u001fƅ�\u003ee������_���Nm�����\u003et�Gu\u001a����Ym\u0013Ҵy��Y�8�c��w��;�u\u0011��=֟�S�w}e/7V���3�\u0019���A�t���Oۀ4S���h\u001d_xK?�#�٪��\u001e(�ʶd���9���n�z��Yu\r}o�|����ɟ�ϋ���f��2�z.��\u001ae�W��\u000c2�Z��s$�h�\u0016~�Am�9_�w�\t�A\u000b�z#�\u001aSSF[�I�\u0016f�隄���g�\u0019q�������liʜ\u0012��\u0007#��#����\n?�m�_Z�#�:�4\u0026|��8^Հ��ܪ����vί�\u0014�\u0014�Ϝ\u0017�+�\\\u001f��׶ �|\u0007����\u003c�+|�\u0005Sfz\u000fU�{�s���_}�guB��\u0008�\u001c�W�W�\u0004�d��jK�\u0013�����A\\�\u003e��\u001b�S�0ϙ����\u0010��\u000c��.tݭm����a\u0008sm;u��c\u0015\u000b)������'\u001dO�T\r-�xi�7L]�z�4�=��\u0026�G���\rZ�!���}e�q-�_=�v�������wi�\u0007�\t?u6ەi��\u003cQ�g�����^�GŇ[�V�Ի��d�W~I��6����\u0017�SF�'����\u001eP1��\u0013\"��Cũ���̘w���m�9\u001f�o΍2N\u001a��w�)�ׯ��\u001eZq)�k�T��g�e�2˻���C\u0016?�S��\u003c���{�����\u000ceB٣ů9-��'������%�z�\u000b_}���z������Ƶxfh\u001b\u0011~��c�\u003c�k|��S���\\��XD�g�\u0003�gΚg���sX�����{'M�S�.�8s~c���\u001e-L˶�\u000fR�{�t����\u0011g�$�g�e���rק_\\�\u0018�fmG�\u001c�p�_�p?y����$}L���P�f�$sVܿz���������\u001e��l\\�2K=Y�i�\u0010~e{��\\�q#�O{t�T�߸\u000b��mDz\u0013~��=I���\u0012\u003e�_L\u0019-��3����sT\u001c�ݰ�Ri�=�'H��ge����Uq)�D=��Ʋ���e�wӍe����v}�}8���y�\\��������\u0026���_}�ҷ�\u0012�WoG�ǹ~}�ί�W�r��es|mG������\u000f�\u000c�7��3_�\u0011_�/H߮_��]�X\u000b���͸�d�Q�_e��9�G\u001b�*̕)�Y{��w\u001by�X(q�=S�3����'(Í�v~�\\��\u001e-N�wRz��{�\u0013?�,��'H�ƥ�~k��i�\u0010��\u0015t��n�n\\��z��\u0017^���{�WOu|�\u0002��C;{f��������_=O��\u0006�gF��v���g�q'�\u000f�K�\u000f��j�\u001a����=Kϼ���\u000b\u0000\u0000��"}]'
```
after some time it looks like there is no more connections left

```
I have no name!@postgres-postgresql-0:/$ psql -U mainflux -d messages
Password for user mainflux:
psql: FATAL:  remaining connection slots are reserved for non-replication superuser connections                                                                             
I have no name!@postgres-postgresql-0:/$ % 

```

```
messages=> SELECT state, usename, backend_type, query, query_start FROM pg_stat_activity;
             state             | usename  |  backend_type  |                                     query                                      |          query_start          
-------------------------------+----------+----------------+--------------------------------------------------------------------------------+-------------------------------
                               |          |                | <insufficient privilege>                                                       | 
                               | postgres |                | <insufficient privilege>                                                       | 
 idle                          | mainflux | client backend | SELECT state, COUNT(*) FROM pg_stat_activity GROUP BY state;                   | 2020-03-03 12:36:15.122162+00
 idle in transaction (aborted) | mainflux | client backend | INSERT INTO messages (id, channel, subtopic, publisher, protocol,             +| 2020-03-03 12:58:30.591874+00
                               |          |                |     name, unit, value, string_value, bool_value, data_value, sum,             +| 
                               |          |                |     time, update_time)                                                        +| 
                               |          |                |     VALUES ($1, $2, $3, $4, $5, $6, $7,                                       +| 
                               |          |                |     $8, $9, $10, $11, $12,                                                    +| 
                               |          |                |     $13, $14);                                                                 | 

```

```
messages=> SELECT state, COUNT(*) FROM pg_stat_activity GROUP BY state;
             state             | count 
-------------------------------+-------
                               |     5
 active                        |     1
 idle                          |     1
 idle in transaction (aborted) |    23
(4 rows)
```

after `idle in transaction` reaches max connections no more connections can be established
now since `data_value` is `text` it is ok that insert fails since we are inserting some  random data in `byte array` abortion should be gracefully handled

### PR DESCRIPTION
**Pull request by mteodor**
### What does this do?
If error happens when trying to save, transaction goes to idle transactions if not released.
Catch the error in code and on error Rolback transaction.

### Which issue(s) does this PR fix/relate to?
Resolves #1055

### CODE REVIEW

File: writers/postgres/messages.go [hunk 1]
```diff
@@ -46,6 +46,19 @@ func (pr postgresRepo) Save(messages ...senml.Message) error {
 	if err != nil {
 		return errors.Wrap(errSaveMessage, err)
 	}
+	defer func() {
```
Comment (manuio) [inline writers/postgres/messages.go:50]: Should this be defined before line 46?

**Review by manuio — CHANGES_REQUESTED** [review]

File: writers/postgres/messages.go [hunk 2]
```diff
@@ -43,6 +43,19 @@ func (pr postgresRepo) Save(messages ...senml.Message) error {
     :time, :update_time);`
 
 	tx, err := pr.db.BeginTxx(context.Background(), nil)
+	defer func() {
+		if err == nil {
```
Comment (manuio) [inline writers/postgres/messages.go]: Can you check `if err != nil` here?

File: writers/postgres/messages.go [hunk 3]
```diff
@@ -43,6 +43,19 @@ func (pr postgresRepo) Save(messages ...senml.Message) error {
...
+			if err = tx.Commit(); err != nil {
+				err = errors.Wrap(errSaveMessage, err)
```
Comment (manuio) [inline writers/postgres/messages.go]: Maybe we need an error for commit

File: writers/postgres/messages.go [hunk 4]
```diff
@@ -43,6 +43,19 @@ func (pr postgresRepo) Save(messages ...senml.Message) error {
...
+			return
+		}
+		err = errors.Wrap(errSaveMessage, err)
+		if txErr := tx.Rollback(); txErr != nil {
+			err = errors.Wrap(errSaveMessage, txErr)
+		}
+		return
+	}()
 	if err != nil {
 		return errors.Wrap(errSaveMessage, err)
```
Comment (manuio) [inline writers/postgres/messages.go:60]: same here

**Review by manuio — CHANGES_REQUESTED** [review]

File: writers/postgres/messages.go [hunk 5]
```diff
@@ -43,6 +43,19 @@ func (pr postgresRepo) Save(messages ...senml.Message) error {
...
+	defer func() {
+		if err == nil {
+			if err = tx.Commit(); err != nil {
+				err = errors.Wrap(errSaveMessage, err)
+			}
+			return
+		}
+		err = errors.Wrap(errSaveMessage, err)
+		if txErr := tx.Rollback(); txErr != nil {
+			err = errors.Wrap(errSaveMessage, txErr)
```
Comment (manuio) [inline writers/postgres/messages.go]: same here

**Review by manuio — CHANGES_REQUESTED** [review]

**Review by drasko — DISMISSED** [review]
LGTM

File: writers/postgres/messages.go [hunk 6]
```diff
@@ -43,6 +44,18 @@ func (pr postgresRepo) Save(messages ...senml.Message) error {
...
+				err = errors.Wrap(err, errors.Wrap(errTransRollback, txErr))
+			}
+			return
+		}
+		if err = tx.Commit(); err != nil {
+			err = errors.Wrap(errSaveMessage, err)
+		}
+		return
+	}()
 	if err != nil {
```
Comment (manuio) [inline writers/postgres/messages.go]: This check should be done before the defer

**Review by manuio — CHANGES_REQUESTED** [review]

**Review by manuio — APPROVED** [review]
LGTM

**Review by blokovi — APPROVED** [review]

**Pull request merged**